specs/_features/eip6110/beacon-chain.md

```diff
@@ -0,0 +1,413 @@
+# EIP-6110 -- The Beacon Chain
+
+## Table of contents
+
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [Constants](#constants)
+  - [Misc](#misc)
+- [Preset](#preset)
+  - [Execution](#execution)
+- [Containers](#containers)
+  - [New containers](#new-containers)
+    - [`DepositReceipt`](#depositreceipt)
+  - [Extended Containers](#extended-containers)
+    - [`ExecutionPayload`](#executionpayload)
+    - [`ExecutionPayloadHeader`](#executionpayloadheader)
+    - [`BeaconState`](#beaconstate)
+- [Beacon chain state transition function](#beacon-chain-state-transition-function)
+  - [Block processing](#block-processing)
+    - [Modified `process_operations`](#modified-process_operations)
+    - [New `get_validator_from_deposit_data`](#new-get_validator_from_deposit_data)
+    - [New `apply_deposit`](#new-apply_deposit)
+    - [New `process_deposit_receipt`](#new-process_deposit_receipt)
+    - [Modified `process_deposit`](#modified-process_deposit)
+    - [Modified `process_execution_payload`](#modified-process_execution_payload)
+- [Testing](#testing)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This is the beacon chain specification of in-protocol deposits processing mechanism.
+This mechanism relies on the changes proposed by [EIP-6110](http://eips.ethereum.org/EIPS/eip-6110).
+
+*Note:* This specification is under development and should be used with care.
+
+## Constants
+
+The following values are (non-configurable) constants used throughout the specification.
+
+### Misc
+
+| Name | Value |
+| - | - |
+| `NOT_SET_DEPOSIT_RECEIPTS_START_INDEX` | `uint64(2**64 - 1)` |
+
+## Preset
+
+### Execution
+
+| Name | Value | Description |
+| - | - | - |
+| `MAX_DEPOSIT_RECEIPTS_PER_PAYLOAD` | `uint64(2**13)` (= 8,192) | Maximum number of deposit receipts allowed in each payload |
```

Comment:
This can only be exceeded if EL gas limit exceeds ~400M, right? (~50k gas per deposit)

Reply:
We have figured out that clever attacker can reduce the cost of deposit down to ~25k. So, this limit can be exceeded with 200M gas block